docs/development-setup.md

```diff
@@ -119,6 +119,18 @@ interface CompositeEntities extends Entities {}
 
 TBD
 
+### XState
```

Comment:
It would be good if you add a basic convention on how the state & events are typed & organized